common/persistence/persistenceInterface.go

```diff
@@ -599,7 +599,7 @@ type (
 	InternalDomainConfig struct {
 		// NOTE: this retention is in days, not in seconds
 		Retention                int32
-		EmitMetric               bool
+		EmitMetric               bool                    // deprecated
```

Comment:
you can completely get rid of this. There is no concern with backward compatibility within the internal server code